eclipse-theia/theia
# core: remove `maximize` with double-click

#### What it does


Fixes: https://github.com/eclipse-theia/theia/issues/5757.

The pull-request removes the problematic and often annoying behavior of when a tabbar is double-click we trigger a `maximized` state.  The behavior was previously problematic as it does not align with vscode, and often we are attempting to dock panels and not maximize them with double-clicks. 

The maximized functionality is still accessible by the command, and keybinding.

#### How to test


1. start the application and open files
2. confirm that double-clicking the tabbar for files, or widgets in the main-area or bottom panel does not maximize them
3. enable the preference `editor.enablePreview`
4. open a file as preview, double clicking the tabbar should dock the file (turn it into a regular editor)
5. confirm the command and keybinding for `view: toggle maximized` works well

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>
